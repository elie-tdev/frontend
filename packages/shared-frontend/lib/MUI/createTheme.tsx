import { createTheme as createMuiTheme } from '@mui/material/styles'

import { makeDefaultTheme } from './defaultTheme'
import { BAIThemeOptions } from './types'
import { createDefaultPaletteOptions } from './utils/palette'
import { createDefaultSpacingOptions } from './utils/spacing'

const createThemeBase = (themeOptions: BAIThemeOptions, isTest = false) => {
  createDefaultSpacingOptions(themeOptions)
  createDefaultPaletteOptions(themeOptions)
  return makeDefaultTheme(createMuiTheme(themeOptions), !isTest)
}

export const createTheme = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions)

export const createThemeTest = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions, true)
