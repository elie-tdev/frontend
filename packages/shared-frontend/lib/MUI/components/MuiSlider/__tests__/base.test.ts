import { MuiSlider } from '../'
import { createThemeTest } from '../../../createTheme'

describe('Slider', () => {
  const theme = createThemeTest()
  const component = MuiSlider(theme)

  it('should be consistent with previous versions: ', () => {
    expect(component).toMatchSnapshot()
  })
})
