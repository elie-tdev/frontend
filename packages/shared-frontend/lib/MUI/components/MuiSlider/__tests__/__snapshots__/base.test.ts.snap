// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should be consistent with previous versions:  1`] = `
Object {
  "styleOverrides": Object {
    "thumb": Object {
      "&:not(.MuiSlider-thumbSizeSmall)": Object {
        "backgroundColor": "#fff",
        "boxShadow": "0px 4px 8px 0px rgba(0, 0, 0, 0.15), inset 0 0 1px 1px #e0e0e0",
      },
      "height": 22,
      "marginLeft": 0,
      "marginTop": 0,
      "width": 22,
    },
    "thumbSizeSmall": Object {
      "&:hover, &.Mui-active": Object {
        "& > .MuiSlider-valueLabelOpen": Object {
          "transform": "scale(75%) translateY(-110%)",
        },
        "transform": "scale(150%) translateY(-35%) translateX(-35%)",
      },
      "&:hover, &.Mui-active, &.Mui-focusVisible": Object {
        "boxShadow": "none",
      },
      "boxShadow": "none",
      "height": 12,
      "transformOrigin": "center",
      "transition": "transform 150ms",
      "width": 12,
    },
    "track": Object {
      "border": "none",
    },
  },
}
`;
